spatialmath/sphere.go

```diff
@@ -200,3 +202,50 @@ func sphereInSphere(inner, outer *sphere) bool {
 func sphereInBox(s *sphere, b *box) bool {
 	return -pointVsBoxDistance(b, s.pose.Point()) >= s.radius
 }
+
+// TODO: add function description
+func (s *sphere) ToPointCloud(options map[string]interface{}) ([]r3.Vector, error) {
+
+	var points [][]float64
+	phi := math.Pi * (3.0 - math.Sqrt(5.0))
+
+	nums := 500 * s.radius
+	for i := 0.; i < nums; i++ {
```

Comment:
Would it be possible to do everything in the same loop?

Reply:
yep, changed it to be in a single for loop